src/sqswatcher/plugins/sge.py

```diff
@@ -84,18 +84,22 @@ def update_cluster(max_cluster_size, cluster_user, update_events, instance_prope
         elif event.action == "ADD":
             hosts_to_add.append(event.host)
 
-    added_hosts = _add_hosts(hosts_to_add, cluster_user)
-    removed_hosts = _remove_hosts(hosts_to_remove)
-
-    succeeded = []
-    failed = []
-    for event in update_events:
-        if event.host.hostname in added_hosts or event.host.hostname in removed_hosts:
-            succeeded.append(event)
-        else:
-            failed.append(event)
-
-    return failed, succeeded
+    try:
+        added_hosts = _add_hosts(hosts_to_add, cluster_user)
+        removed_hosts = _remove_hosts(hosts_to_remove)
+
+        succeeded = []
+        failed = []
+        for event in update_events:
+            if event.host.hostname in added_hosts or event.host.hostname in removed_hosts:
+                succeeded.append(event)
+            else:
+                failed.append(event)
+
+        return failed, succeeded
+    except Exception as e:
```

Comment:
what do you think if we move this to the sqswatcher.py where we call update_cluster so that we have a uniform behaviour for all schedulers without duplication? Otherwise it LGTM